src/App.tsx

```diff
@@ -1,24 +1,220 @@
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import {
+  CountPerStatus,
+  getCountPerStatus,
+  getVisibleTodos,
+} from './utils/functions';
+import { FilteringMethod } from './types/FilteringStatus';
 
-const USER_ID = 0;
+const USER_ID = 6648;
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filterMethod, setFilterStatus] = useState(FilteringMethod.ALL);
+  const [loadingTodosIDs, setLoadingTodosIDs] = useState<number[]>([]); // problem is here
+  const [textFieldValue, setTextFieldValue] = useState('');
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.NONE);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const todosFromServer = await getTodos(USER_ID);
+
+        setTodos(todosFromServer);
+      } catch {
+        setErrorMessage(ErrorMessage.LOAD);
+
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.NONE);
+        }, 3000);
+      }
+    })();
+  }, []);
+
+  const showErrorMessage = useCallback((message: ErrorMessage) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage(ErrorMessage.NONE);
+    }, 3000);
+  }, []);
+
+  const setTodoLoading = useCallback((id: number) => {
+    setLoadingTodosIDs(prevState => [...prevState, id]);
+  }, []);
+
+  const unsetTodoLoading = useCallback((id: number) => {
+    setLoadingTodosIDs(prevState => prevState.filter(num => num !== id));
+  }, []);
+
+  const addTodo = useCallback(async () => {
+    setIsInputDisabled(true);
+
+    const newTodo = {
+      title: textFieldValue,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({
+      ...newTodo,
+      id: 0,
+    });
+
+    try {
+      const resultingTodo = await postTodo(newTodo);
+
+      setTodos(prev => [
+        ...prev,
+        resultingTodo,
+      ]);
+      setTextFieldValue('');
+    } catch {
+      showErrorMessage(ErrorMessage.ADD);
+    } finally {
+      setIsInputDisabled(false);
+      setTempTodo(null);
+    }
+  }, [textFieldValue]);
+
+  const deleteTodo = useCallback(async (id: number) => {
+    setTodoLoading(id);
+
+    try {
+      await removeTodo(id);
+
+      const newTodosList = todos.filter(todo => todo.id !== id);
+
+      setTodos(() => newTodosList);
+    } catch {
+      showErrorMessage(ErrorMessage.DELETE);
+    } finally {
+      unsetTodoLoading(id);
+    }
+  }, [todos]);
+
+  const deleteAllCompleted = useCallback(async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    completedTodos.forEach(todo => {
+      setTodoLoading(todo.id);
+    });
```

Comment:
it's nice that you've reused this function, but wouldn't it be more logical to map completedTodos to get their ids and update state once, adding all ids to state

Reply:
It's not critical, just a suggestion